src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+                    // Do not default to no_task_metrics on Worker.
+                    if self.attack_mode == GooseMode::Worker {
+                        false
+                    } else {
+                        default
+                    }
+                } else {
+                    false
+                };
+        }
+    }
+
+    // Determine if the no_metrics flag is enabled.
+    fn set_no_metrics(&mut self) {
+        // Overload self.configuration.no_metrics so it's available to Users.
+        if !self.configuration.no_metrics {
+            self.configuration.no_metrics = if let Some(default) = self.defaults.no_metrics {
+                // Do not default to no_metrics on Worker.
+                if self.attack_mode == GooseMode::Worker {
+                    false
+                } else {
+                    default
+                }
+            } else {
+                false
+            };
+        }
+    }
+
+    // Determine if the sticky_follow flag is enabled.
+    fn set_sticky_follow(&mut self) {
+        // Overload self.configuration.sticky_follow so it's available to Users.
+        if !self.configuration.sticky_follow {
+            self.configuration.sticky_follow = if let Some(default) = self.defaults.sticky_follow {
+                // Do not default to sticky_follow on Worker.
+                if self.attack_mode == GooseMode::Worker {
+                    false
+                } else {
+                    default
+                }
+            } else {
+                false
+            };
+        }
+    }
+
+    #[cfg(feature = "gaggle")]
+    // Determine if no_hash_check is enabled.
+    fn no_hash_check(&self) -> bool {
+        if self.configuration.no_hash_check {
+            true
+        } else if let Some(default) = self.defaults.no_hash_check {
+            // Do not default to no_hash_check on Worker.
+            if self.attack_mode == GooseMode::Worker {
+                false
+            } else {
+                default
+            }
+        } else {
+            false
+        }
+    }
+
+    // If enabled, returns the path of the metrics_file, otherwise returns None.
+    fn get_metrics_file_path(&mut self) -> Result<Option<&str>, GooseError> {
+        // If metrics are disabled, or running in Manager mode, there is no
+        // metrics file, exit immediately.
+        if self.configuration.no_metrics || self.attack_mode == GooseMode::Manager {
+            return Ok(None);
+        }
+
+        // If --metrics-file is set, return it.
+        if !self.configuration.metrics_file.is_empty() {
+            return Ok(Some(&self.configuration.metrics_file));
+        }
+
+        // If GooseDefault::MetricFile is set, return it.
+        if let Some(default_metrics_file) = &self.defaults.metrics_file {
+            return Ok(Some(default_metrics_file));
+        }
+
+        // Otherwise there is no metrics file.
+        Ok(None)
+    }
+
+    // Configure metrics log format.
```

Comment:
@LionsAd Here we're overloading `configuration` again, as we need this on Workers.